core/src/main/java/org/mule/runtime/core/internal/exception/GlobalErrorHandler.java

```diff
@@ -41,56 +32,61 @@ public Publisher<CoreEvent> apply(Exception exception) {
 
   @Override
   public void initialise() throws InitialisationException {
-    setFromGlobalErrorHandler();
-    super.initialise();
-  }
-
-  private void setFromGlobalErrorHandler() {
-    this.getExceptionListeners().stream()
-        .filter(exceptionListener -> exceptionListener instanceof TemplateOnErrorHandler)
-        .forEach(exceptionListener -> ((TemplateOnErrorHandler) exceptionListener).setFromGlobalErrorHandler(true));
+    if (featureFlaggingService.isEnabled(REUSE_GLOBAL_ERROR_HANDLER)) {
+      if (!initialised.getAndSet(true)) {
```

Comment:
When the feature flag is enabled, the global error handler will be used for all the flows instead of creating local copies. This make the lifecycle phases to be called one time for each flow, so we need to add this atomic boolean to only call these phases one time in the corresponding exception listeners.

Reply:
The way is working now the exception listener phases are managed by the flow but yes, I think it should be investigated a better way to do this.